src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -30,178 +31,175 @@ import { WithDetailsReceipt } from '@/components/WithDetailsReceipt';
 
 import { calculateCanViewAccount, calculateIsAccountViewOnly } from '@/state/accountCalculators';
 import { getSubaccount } from '@/state/accountSelectors';
-import { useAppSelector } from '@/state/appTypes';
-import { getUserVault } from '@/state/vaultSelectors';
-
+import { useAppDispatch, useAppSelector } from '@/state/appTypes';
+import { getVaultAccount, getVaultForm } from '@/state/vaultSelectors';
+import {
+  setVaultFormAmount,
+  setVaultFormConfirmationStep,
+  setVaultFormOperation,
+  setVaultFormSlippageAck,
+} from '@/state/vaults';
+
+import { assertNever } from '@/lib/assertNever';
 import { MustBigNumber } from '@/lib/numbers';
-
-type VaultFormError = {
-  type: AlertType;
-  key: string;
-  short?: string;
-  long?: React.ReactNode;
-};
+import { safeAssign } from '@/lib/objectHelpers';
 
 // errors we don't want to show aggressive visual cues about, just disable submit
-const lightErrorKeys = new Set(['disconnected', 'view-only', 'amount-empty']);
-
-const SLIPPAGE_PERCENT_WARN = 0.01;
-const SLIPPAGE_PERCENT_ACK = 0.01;
+const lightErrorKeys = new Set<string>([
+  'ACCOUNT_DATA_MISSING',
+  'AMOUNT_EMPTY',
+] satisfies VaultFormValidationErrorType['name'][]);
 
 type VaultDepositWithdrawFormProps = {
   initialType?: 'deposit' | 'withdraw';
```

Comment:
Would it be helpful if `initialType` matched `vaultForm.operation`?

Reply:
would save a line @`90`

Reply:
good call